### PR TITLE
build: update remark-parse to v8

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] |
| :--------------------: |

## 🧰 Changes

Updates remark-parse to v8!

We realized that we incorrect about which version of remark-parse is the complete rewrite, it's v9 not v8. And, mdx@v1 uses remark-parse@v8! So we don't have to do our rewrite yet.

This updates and fixes some _weird_ issues.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -137,13 +137,13 @@ export function plain(text, opts = {}, components = {}) {
...
-      components,
-    })
-    .processSync(text).contents;
+  const proc = htmlProcessor(opts).use(rehypeReact, {
+    createElement: React.createElement,
+    Fragment: React.Fragment,
+    components,
+  });
+
+  return proc.stringify(proc.runSync(proc.parse(text)));
```
Comment (kellyjosephprice) [inline index.js:146]: These _should_ be equivalent, but it was returning markdown rather than react?!

File: jest.config.js [hunk 2]
```diff
@@ -33,4 +32,4 @@ const unit = {
   },
 };
 
-module.exports = { projects: [unit, browser] };
+module.exports = { projects: [unit] };
```
Comment (kellyjosephprice) [inline jest.config.js:35]: Slowly deleting the browser tests.

File: processor/parse/compact-headings.js [hunk 3]
```diff
@@ -6,8 +6,6 @@ function tokenizer(eat, value) {
   const [match, hash, text] = rgx.exec(value);
 
   const now = eat.now();
-  now.column += match.length;
-  now.offset += match.length;
```
Comment (kellyjosephprice) [inline processor/parse/compact-headings.js:10]: This looks like it was incorrect before.

File: processor/parse/flavored/code-tabs.js [hunk 4]
```diff
@@ -49,8 +48,8 @@ function parser() {
   const tokenizers = Parser.prototype.blockTokenizers;
   const methods = Parser.prototype.blockMethods;
 
-  tokenizers.CodeTabs = tokenizer;
-  methods.splice(methods.indexOf('newline'), 0, 'CodeTabs');
+  tokenizers.codeTabs = tokenizer;
```
Comment (kellyjosephprice) [inline processor/parse/flavored/code-tabs.js:51]: Not required, but more conventional.

File: index.js (hunk 1 above)
Comment (rafegoldberg) [reply index.js:146]: Hmmm, this kinda stresses me out… Why was it returning the wrong thing? Will this have adverse effects on the ReadMe-side render?

Comment (kellyjosephprice) [reply index.js:146]: Only, if someone is grabbing the processor and calling `.processSync` directly? But, yea, maybe I should investigate more.

**Review by erunion — APPROVED** [review]
i don't know much about the differences in the new version of remark but tests are good so 👍 

did notice that the CI workflow here is still set up to test node 14 so you can probably update that to just do node 16. and maybe node 18 and 20 too?

File: index.js (hunk 1 above)
Comment (rafegoldberg) [reply index.js:146]: True, just seems…I dunno…seems sketch that this would suddenly start returning something different???

Comment (kellyjosephprice) [reply index.js]: Ah, it was an intentional change, probably for `rehype-react`. If it's returning an object, it using `.result`, so it's not overloading the value of `.contents`: https://github.com/unifiedjs/unified/commit/c3ba172708e719baa963ee5826a7c81774a2784e

**Comment by kellyjosephprice** [comment]
> i don't know much about the differences in the new version of remark but tests are good so 👍
> 
> did notice that the CI workflow here is still set up to test node 14 so you can probably update that to just do node 16. and maybe node 18 and 20 too?

Looks like node 20 is erroring due to `node-sass`. I'm assuming we should hold off on that till we decide what to use in the main app?

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.64.0</b>](/readmeio/markdown/releases/tag/v6.64.0)
<br>

**Pull request merged**